lib/data/datasource/auth_data_source.dart

```diff
@@ -1,7 +1,37 @@
 import '../model/request/login_request.dart';
 import '../model/response/base_response.dart';
 import '../model/response/login_response.dart';
+import '../api/auth_api.dart';
 
 abstract class AuthDataSource {
   Future<BaseResponse<LoginData>> login(LoginRequest request);
-}
\ No newline at end of file
+  Future<BaseResponse<void>> logout();
+  Future<BaseResponse<LoginData>> reissue();
+  Future<BaseResponse<void>> withdrawal();
+}
+
+class AuthDataSourceImpl extends AuthDataSource {
+  final AuthApi _authApi;
+
+  AuthDataSourceImpl(this._authApi);
+
+  @override
+  Future<BaseResponse<LoginData>> login(LoginRequest request) async {
+    return await _authApi.login(request);
+  }
+
```

Comment:
C:
```suggestion
  @override
  Future<BaseResponse<LoginData>> login(LoginRequest request) {
    return _authApi.login(request);
  }
```

DataSource단에서는 전부다 await 빼도 될 듯